go/cmd/remotetool/main.go

```diff
@@ -0,0 +1,72 @@
+// Main package for the remotetool binary.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/bazelbuild/remote-apis-sdks/go/pkg/tool"
+
+	rflags "github.com/bazelbuild/remote-apis-sdks/go/pkg/flags"
+	log "github.com/golang/glog"
+)
+
+// OpType denotes the type of operation to perform.
+type OpType string
+
+const (
+	downloadActionResult OpType = "download_action_result"
+)
+
+var (
+	operation    = flag.String("operation", "download_action_result", "Specifies the operation to perform. Currently only get_action_result is supported.")
+	actionDigest = flag.String("action_digest", "", "Action digest in <digest/size_bytes> format.")
+	pathPrefix   = flag.String("path", "/tmp", "Directory to which outputs should be downloaded to. Defaults to /tmp/.")
+)
+
+func validate() {
+	if *actionDigest == "" {
+		log.Exitf("action_digest must be specified.")
+	}
+	if _, err := os.Stat(*pathPrefix); err != nil && os.IsNotExist(err) {
+		log.Exitf("path %v does not exist.", *pathPrefix)
+	}
```

Comment:
shouldn't this validation happen depending on the operation?

Reply:
Actually, I realized path doesn't have to be validated. The SDK creates it if it doesn't exist, so removed it. But generally also agree that this validation should happen depending on the type of operation.